app/src/adapters/web3LibraryAdapter.ts

```diff
@@ -0,0 +1,61 @@
+import { BrowserProvider /*, Signer */ } from 'ethers'
+
+// Define interface for web3 library
+export interface IWeb3Library {
+  initialize(): void
+  connectWallet(): Promise<void>
+  requestSign(message: string): Promise<string>
+  getAddress(): Promise<string>
+}
+
+// Adapter for ethers.js
+export class EthersJsAdapter implements IWeb3Library {
+  private provider: any
+  //private provider: ethers.providers.Web3Provider | null = null;
+  //private signer: ethers.Signer | null = null;
+  private signer: any
+
+  initialize(): void {
+    // Initialize provider
+    if ('ethereum' in window) this.provider = new BrowserProvider(window.ethereum as any)
+    //this.signer = this.provider.getSigner();
```

Comment:
maybe u can fix like lint problem with this
```typescript
(window as any).ethereum
```

Reply:
Yeah, this passed the CI/CD test though. Initially it was failing because I was using `window.ethereum` direct but I changed to `window.ethereum as any` in `BrowserProvider` constructor which is similar to what you're suggesting.